logconfig/config.go

```diff
@@ -18,88 +18,113 @@
 package logconfig
 
 import (
+	"fmt"
+	"io"
+	stdlog "log"
 	"os"
+	"path"
 	"strconv"
 	"strings"
 	"time"
 
+	"github.com/arthurkiller/rollingwriter"
+	"github.com/mysteriumnetwork/go-openvpn/openvpn"
+	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
 	"github.com/rs/zerolog/log"
 )
 
-// TODO remove this after log rotation is re-done on zerolog
-// leaving for the reference
-//const seewayLogXMLConfigTemplate = `
-//<seelog minlevel="{{.LogLevel}}">
-//	<outputs formatid="main">
-//		<console/>
-//		{{ if (ne .Filepath "") }}
-//		<rollingfile
-//			formatid="main"
-//			filename="{{.Filepath}}"
-//			maxrolls="7"
-//			type="date"
-//			datepattern="2006.01.02"
-//		/>
-//		{{ end }}
-//	</outputs>
-//	<formats>
-//		<format id="main" format="%UTCDate(2006-01-02T15:04:05.999999999) [%Level] %Msg%n"/>
-//	</formats>
-//</seelog>
-//`
+const (
+	timestampFmt = "2006-01-02T15:04:05.000"
+)
 
-// BootstrapWith loads log package into the overall system
-func BootstrapWith(opts *LogOptions) {
-	if opts != nil {
-		CurrentLogOptions = *opts
+// Bootstrap configures logger defaults (console)
+func Bootstrap() {
+	var trimPrefixes = []string{
+		"/vendor",
+		"/go/pkg/mod",
 	}
-	configureZerolog(opts)
-	log.Info().Msgf("Log level: %s", CurrentLogOptions.LogLevel)
-	if CurrentLogOptions.Filepath != "" {
-		log.Info().Msg("Log file path: " + CurrentLogOptions.Filepath)
+	cwd, _ := os.Getwd()
+	zerolog.TimeFieldFormat = time.RFC3339Nano
+	zerolog.CallerMarshalFunc = func(file string, line int) string {
+		relFile := strings.TrimPrefix(file, cwd)
+		for i := range trimPrefixes {
+			relFile = trimLeftInclusive(relFile, trimPrefixes[i])
+		}
+		return fmt.Sprintf("%-41v", relFile+":"+strconv.Itoa(line))
 	}
-}
 
-// Bootstrap loads log package into the overall system with debug defaults
-func Bootstrap() {
-	BootstrapWith(
-		&LogOptions{
-			LogLevel: zerolog.DebugLevel,
-		},
-	)
+	openvpn.UseLogger(zerologOpenvpnLogger{})
+	logger := makeLogger(consoleWriter())
+	setGlobalLogger(&logger)
 }
 
-func padRight(s string, length int) string {
-	if len(s) >= length {
-		return s
+// Configure configures logger using app config (console + file, level)
+func Configure(opts *LogOptions) {
+	CurrentLogOptions = *opts
+	log.Info().Msgf("Log level: %s", opts.LogLevel)
+	if opts.Filepath != "" {
+		log.Info().Msg("Log file path: " + opts.Filepath)
```

Comment:
Try to be consistant. Use Msgf it for formatted messages.

```go
log.Info().Msgf("Log level: %s", opts.LogLevel) // Ok
log.Info().Msg("Log file path: " + opts.Filepath) 
```

Reply:
Fixed. Thanks.